openmdao/core/problem.py

```diff
@@ -138,8 +144,30 @@ def __init__(self, model=None, driver=None, comm=None, name=None, **options):
         """
         Initialize attributes.
         """
+        global _problem_counter
```

Comment:
Do you also need to global `_problem_names ` since you are appending to it?

Reply:
Good point. Fixed.